data-serving/data-service/src/controllers/case.ts

```diff
@@ -61,113 +64,160 @@ export class CasesController {
             return;
         }
 
+        logger.info(`Streaming case data in format ${req.body.format} matching query ${req.body.query} for correlation ID ${req.body.correlationId}`);
+
         const queryLimit = Number(req.body.limit);
 
+        logger.info(`Request body: ${req.body}`);
+
         // Goofy Mongoose types require this.
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        let casesQuery: any;
+        let cursor: QueryCursor<CaseDocument>;
         try {
+            // Stream from mongoose directly into response
+            // Use provided query and limit, if provided
             if (req.body.query) {
-                casesQuery = casesMatchingSearchQuery({
+                logger.info('Request body with query');
+                cursor = casesMatchingSearchQuery({
                     searchQuery: req.body.query as string,
                     count: false,
-                });
+                    limit: queryLimit,
+                })
+                .cursor();
+            } else if (req.body.caseIds && queryLimit) {
+                logger.info('Request body with case IDs and limit');
+                cursor = Case.find({
+                    _id: { $in: req.body.caseIds },
+                })
+                .lean()
+                .limit(queryLimit)
+                .collation({
+                    locale: 'en_US',
+                    strength: 2,
+                })
+                .cursor();
             } else if (req.body.caseIds) {
-                casesQuery = Case.find({
+                logger.info('Request body with case IDs and no limit');
+                cursor = Case.find({
                     _id: { $in: req.body.caseIds },
-                }).lean();
+                })
+                .lean()
+                .collation({
+                    locale: 'en_US',
+                    strength: 2,
+                })
+                .cursor();
+            } else if (queryLimit) {
+                logger.info('Request body with limit and no case IDs');
+                cursor = Case.find({ list: true })
+                .lean()
+                .limit(queryLimit)
+                .collation({
+                    locale: 'en_US',
+                    strength: 2,
+                })
+                .cursor();
             } else {
-                casesQuery = Case.find({ list: true }).lean();
-            }
-
-            // Limit number of results if present
-            // eslint-disable-next-line
-            if (queryLimit) {
-                casesQuery = casesQuery.limit(queryLimit);
+                logger.info('Request body with no query, limit, or case IDs');
+                cursor = Case.find({ list: true })
```

Comment:
Fine with this in data service, but curator service shouldn't hit this and redirect to the full dataset download

Reply:
For requests for .csv format? Or including .tsv and .json?

Reply:
Yes, for csv. I don't think we should allow full dataset downloads except the standard csv since it'll put unnecessary load on data service